.jenkins/test.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash -f -e
+
+##################################################
+# functions
+##################################################
+
+exitError()
+{
+    echo "ERROR $1: $3" 1>&2
+    echo "ERROR     LOCATION=$0" 1>&2
+    echo "ERROR     LINE=$2" 1>&2
+    exit $1
+}
+
+showUsage()
+{
+    echo "usage: `basename $0` [-h]"
+    echo ""
+    echo "optional arguments:"
+    echo "-h           show this help message and exit"
+}
+
+parseOptions()
+{
+    # process command line options
+    while getopts ":c:dfhl" opt
+    do
+        case $opt in
+        h) showUsage; exit 0 ;;
+        \?) showUsage; exitError 301 ${LINENO} "invalid command line option (-${OPTARG})" ;;
+        :) showUsage; exitError 302 ${LINENO} "command line option (-${OPTARG}) requires argument" ;;
+        esac
+    done
+
+}
+
+# echo basic setup
+echo "####### executing: $0 $* (PID=$$ HOST=$HOSTNAME TIME=`date '+%D %H:%M:%S'`)"
+
+# start timer
+T="$(date +%s)"
+
+# parse command line options (pass all of them to function)
+parseOptions $*
+
+# check presence of env directory
+pushd `dirname $0` > /dev/null
+envloc=`/bin/pwd`
+popd > /dev/null
+
+# Download the env
+. ${envloc}/env.sh
+
+# setup module environment and default queue
+if [ ! -f ${envloc}/env/machineEnvironment.sh ] ; then
+    echo "Error 1201 test.sh ${LINENO}: Could not find ${envloc}/env/machineEnvironment.sh"
+    exit 1
+fi
+. ${envloc}/env/machineEnvironment.sh
+
+# load machine dependent environment
+if [ ! -f ${envloc}/env/env.${host}.sh ] ; then
+    exitError 1202 ${LINENO} "could not find ${envloc}/env/env.${host}.sh"
+fi
+. ${envloc}/env/env.${host}.sh
+
+
+# set root and move to testuite folder
+root=`pwd`
```

Comment:
what is not ran from root? does a `script_path=$(dirname $(which $0))` make sense?